abseil build with clang failing due to '-ignore:4221'

**Describe the bug**
Linking against abseil while using clang (not clang-cl) in windows yields the error 

`clang : error : unknown argument: '-ignore:4221'`

The file C:\vcpkg\installed\x64-windows\share\absl\abslTargets.cmake has 

```
set_target_properties(absl::log_severity PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "absl::core_headers;\$<LINK_ONLY:-ignore:4221>"
)
```

on each target (see the `<LINK_ONLY:-ignore:4221>` at the end).

The abseil source code file

[](https://github.com/abseil/abseil-cpp/blob/master/absl/copts/AbseilConfigureCopts.cmake) applies `${ABSL_MSVC_LINKOPTS}`
in an appropriate way for MSVC but not for WIN32.

The variable ABSL_MSVC_LINKOPTS is defined in [](https://github.com/abseil/abseil-cpp/blob/master/absl/copts/GENERATED_AbseilCopts.cmake),

```
list(APPEND ABSL_MSVC_LINKOPTS
    "-ignore:4221"
)
```

**Environment**
- OS: [e.g. Windows/Linux etc...]
- Compiler: revision
Windows 10, clang 10.0.0 (not clang-cl)

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install absl
2. See error
Simply link against absl with cmake and build with clang.exe.  I've created a simple reproducer here, which assumes clang.exe is in the default installed location. https://github.com/jjpepper/vcpkg-absl-clang-problem

**Expected behavior**
I expect the sample program to compile and link when built in `x64-Clang-Release` configuration.  It is just a minimal program

`int main() { return 0; }`



**Failure logs** 
```
>------ Build All started: Project: vcpkg-absl-clang-problem, Configuration: x64-Clang-Release ------
  [1/2] Building CXX object CMakeFiles/main.dir/main.cpp.obj
  [2/2] Linking CXX executable main.exe
  FAILED: main.exe 
  cmd.exe /C "cd . && C:\PROGRA~1\LLVM\bin\clang.exe -fuse-ld=lld-link -nostartfiles -nostdlib -m64 -fdiagnostics-absolute-paths -O2 -g -DNDEBUG -Xclang -gcodeview -D_DLL -D_MT -Xclang --dependent-lib=msvcrt   CMakeFiles/main.dir/main.cpp.obj  -o main.exe -Xlinker /implib:main.lib -Xlinker /pdb:main.pdb -Xlinker /version:0.0   C:/vcpkg/installed/x64-windows/lib/absl_hash.lib  C:/vcpkg/installed/x64-windows/lib/absl_bad_variant_access.lib  C:/vcpkg/installed/x64-windows/lib/absl_city.lib  C:/vcpkg/installed/x64-windows/lib/absl_raw_hash_set.lib  C:/vcpkg/installed/x64-windows/lib/absl_bad_optional_access.lib  C:/vcpkg/installed/x64-windows/lib/absl_hashtablez_sampler.lib  C:/vcpkg/installed/x64-windows/lib/absl_exponential_biased.lib  C:/vcpkg/installed/x64-windows/lib/absl_synchronization.lib  C:/vcpkg/installed/x64-windows/lib/absl_graphcycles_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_stacktrace.lib  C:/vcpkg/installed/x64-windows/lib/absl_symbolize.lib  C:/vcpkg/installed/x64-windows/lib/absl_malloc_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_debugging_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_demangle_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_time.lib  C:/vcpkg/installed/x64-windows/lib/absl_strings.lib  C:/vcpkg/installed/x64-windows/lib/absl_strings_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_int128.lib  C:/vcpkg/installed/x64-windows/lib/absl_base.lib  C:/vcpkg/installed/x64-windows/lib/absl_dynamic_annotations.lib  C:/vcpkg/installed/x64-windows/lib/absl_spinlock_wait.lib  C:/vcpkg/installed/x64-windows/lib/absl_civil_time.lib  C:/vcpkg/installed/x64-windows/lib/absl_time_zone.lib  C:/vcpkg/installed/x64-windows/lib/absl_throw_delegate.lib  C:/vcpkg/installed/x64-windows/lib/absl_raw_logging_internal.lib  C:/vcpkg/installed/x64-windows/lib/absl_log_severity.lib  -ignore:4221  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -loldnames && cmd.exe /C "cd /D C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release && powershell -noprofile -executionpolicy Bypass -file C:/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Users/jesse/source/repos/github/jjpepper/vcpkg-absl-clang-problem/out/build/x64-Clang-Release/main.exe -installedDir C:/vcpkg/installed/x64-windows/bin -OutVariable out""
C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\clang : error : unknown argument: '-ignore:4221'
  ninja: build stopped: subcommand failed.

Build All failed.
```

**Additional context**
Add any other context about the problem here.

When I try with the same generator as you, I get the following error:

```
1> [CMake] -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
1> [CMake] CMake Error at C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\CMakeLists.txt:2 (project):
1> [CMake]   Failed to run MSBuild command:
1> [CMake] 
1> [CMake]     C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/MSBuild/Current/Bin/MSBuild.exe
1> [CMake] 
1> [CMake]   to get the value of VCTargetsPath:
1> [CMake] 
1> [CMake]     Microsoft (R) Build Engine version 16.7.0+b89cb5fde for .NET Framework
1> [CMake]     Copyright (C) Microsoft Corporation. All rights reserved.
1> [CMake]     
1> [CMake]     Build started 22/10/2020 4:54:14 PM.
1> [CMake]     Project "C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\CMakeFiles\3.17.20032601-MSVC_2\VCTargetsPath.vcxproj" on node 1 (default targets).
1> [CMake]     C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(411,5): error MSB8020: The build tools for ClangCl (Platform Toolset = 'ClangCl') cannot be found. To build using the ClangCl build tools, please install ClangCl build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\CMakeFiles\3.17.20032601-MSVC_2\VCTargetsPath.vcxproj]
1> [CMake]     Done Building Project "C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\CMakeFiles\3.17.20032601-MSVC_2\VCTargetsPath.vcxproj" (default targets) -- FAILED.
1> [CMake]     
1> [CMake]     Build FAILED.
1> [CMake]     
1> [CMake]     "C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\CMakeFiles\3.17.20032601-MSVC_2\VCTargetsPath.vcxproj" (default target) (1) ->
1> [CMake]     (PrepareForBuild target) -> 
1> [CMake]       C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(411,5): error MSB8020: The build tools for ClangCl (Platform Toolset = 'ClangCl') cannot be found. To build using the ClangCl build tools, please install ClangCl build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\jesse\source\repos\github\jjpepper\vcpkg-absl-clang-problem\out\build\x64-Clang-Release\CMakeFiles\3.17.20032601-MSVC_2\VCTargetsPath.vcxproj]
1> [CMake]     
1> [CMake]         0 Warning(s)
1> [CMake]         1 Error(s)
1> [CMake]     
1> [CMake]     Time Elapsed 00:00:00.17
1> [CMake]     
1> [CMake] 
1> [CMake]   Exit code: 1
1> [CMake] 
1> [CMake] 
1> [CMake] 
1> [CMake] -- Configuring incomplete, errors occurred!
```

My vcpkg is 8 days old:
```
$ git log
commit bfe983d721a0f2e0cd492f4b5632a952072b7e93 (HEAD -> master, origin/master, origin/HEAD)
Author: Billy O'Neal <bion@microsoft.com>
Date:   Wed Oct 14 17:38:40 2020 -0700
```

I tried updating to the latest vcpkg and rebuilding, but the problem persists.

On top of that, I get a different new error in with clang-cl.exe when I add `#include <absl/container/flat_hash_map.h>`, so after hours of rebuilding, now my clang-cl build is broken too. :-(

```
  [1/2] Building CXX object CMakeFiles\main.dir\main.cpp.obj
  FAILED: CMakeFiles/main.dir/main.cpp.obj 
  C:\PROGRA~1\LLVM\bin\clang-cl.exe  /nologo -TP  -IC:\vcpkg\installed\x64-windows\include -m64 -fdiagnostics-absolute-paths  /DWIN32 /D_WINDOWS /GR /EHsc /Zi /O2 /Ob1 /DNDEBUG -MD   -std:c++17 /showIncludes /FoCMakeFiles\main.dir\main.cpp.obj /FdCMakeFiles\main.dir\ -c ..\..\..\main.cpp
  In file included from ..\..\..\main.cpp:1:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/container/flat_hash_map.h:41:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/container/internal/raw_hash_map.h:24:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/container/internal/raw_hash_set.h:113:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/container/internal/hashtablez_sampler.h:50:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/synchronization/mutex.h:71:
  In file included from C:\vcpkg\installed\x64-windows\include\absl/synchronization/internal/kernel_timeout.h:32:
C:\vcpkg\installed\x64-windows\include\absl\base\internal\raw_logging.h(173,10): error : an attribute list cannot appear here
  ABSL_DLL ABSL_INTERNAL_ATOMIC_HOOK_ATTRIBUTES extern base_internal::AtomicHook<
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  C:\vcpkg\installed\x64-windows\include\absl\base\internal\atomic_hook.h(49,46): note: expanded from macro 'ABSL_INTERNAL_ATOMIC_HOOK_ATTRIBUTES'
  #define ABSL_INTERNAL_ATOMIC_HOOK_ATTRIBUTES ABSL_CONST_INIT
                                               ^~~~~~~~~~~~~~~
  C:\vcpkg\installed\x64-windows\include\absl\base\attributes.h(616,25): note: expanded from macro 'ABSL_CONST_INIT'
  #define ABSL_CONST_INIT [[clang::require_constant_initialization]]
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
  ninja: build stopped: subcommand failed.
```

--- a/ports/abseil/fix-msvc-flags.patch
+++ b/ports/abseil/fix-msvc-flags.patch
@@ -1,0 +1,12 @@
+diff --git a/absl/copts/AbseilConfigureCopts.cmake b/absl/copts/AbseilConfigureCopts.cmake
+index acd46d0..6081527 100644
+--- a/absl/copts/AbseilConfigureCopts.cmake
++++ b/absl/copts/AbseilConfigureCopts.cmake
+@@ -41,7 +41,6 @@ elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+     # clang-cl is half MSVC, half LLVM
+     set(ABSL_DEFAULT_COPTS "${ABSL_CLANG_CL_FLAGS}")
+     set(ABSL_TEST_COPTS "${ABSL_CLANG_CL_FLAGS};${ABSL_CLANG_CL_TEST_FLAGS}")
+-    set(ABSL_DEFAULT_LINKOPTS "${ABSL_MSVC_LINKOPTS}")
+   else()
+     set(ABSL_DEFAULT_COPTS "${ABSL_LLVM_FLAGS}")
+     set(ABSL_TEST_COPTS "${ABSL_LLVM_FLAGS};${ABSL_LLVM_TEST_FLAGS}")

--- a/ports/abseil/portfile.cmake
+++ b/ports/abseil/portfile.cmake
@@ -8,12 +8,15 @@ vcpkg_from_github(
     REF 0f3bb466b868b523cf1dc9b2aaaed65c77b28862 #LTS 20200923, Patch 2
     SHA512 17e766a2f7a655a3877eb3accc5745a910b69a5e2426b7ce7f6d31095523dd32d48a709c5f8380488b4cb93ce9faadedc08f0481dbdbd00cf68831541d724b4d
     HEAD_REF master
-    # in C++17 mode, use std::any, std::optional, std::string_view, std::variant
-    # instead of the library replacement types
-    # in C++11 mode, force use of library replacement types, otherwise the automatic
-    # detection can cause ABI issues depending on which compiler options
-    # are enabled for consuming user code
-    PATCHES fix-cxx-standard.patch
+    PATCHES
+        # in C++17 mode, use std::any, std::optional, std::string_view, std::variant
+        # instead of the library replacement types
+        # in C++11 mode, force use of library replacement types, otherwise the automatic
+        # detection can cause ABI issues depending on which compiler options
+        # are enabled for consuming user code
+        fix-cxx-standard.patch
+        # Official patch https://github.com/abseil/abseil-cpp/commit/58a9c6d53f93078101c2c0bd98d2951e74328a55
+        fix-msvc-flags.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/abseil/vcpkg.json
+++ b/ports/abseil/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "abseil",
   "version-string": "2020-09-23",
-  "port-version": 1,
+  "port-version": 2,
   "description": [
     "an open-source collection designed to augment the C++ standard library.",
     "Abseil is an open-source collection of C++ library code designed to augment the C++ standard library. The Abseil library code is collected from Google's own C++ code base, has been extensively tested and used in production, and is the same code we depend on in our daily coding lives.",

--- a/versions/a-/abseil.json
+++ b/versions/a-/abseil.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1558fdc71dd0f0988cd2a53ba439bed3cc899247",
+      "version-string": "2020-09-23",
+      "port-version": 2
+    },
+    {
       "git-tree": "3d9d65e41e314658c5e18839117ebf7f060b9cb7",
       "version-string": "2020-09-23",
       "port-version": 1

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -10,7 +10,7 @@
     },
     "abseil": {
       "baseline": "2020-09-23",
-      "port-version": 1
+      "port-version": 2
     },
     "absent": {
       "baseline": "0.3.1",

